include/RAJA/policy/openmp/forall.hpp

```diff
@@ -66,12 +67,16 @@ RAJA_INLINE void forall_impl(const omp_parallel_exec<InnerPolicy>&,
                              Iterable&& iter,
                              Func&& loop_body)
 {
-#pragma omp parallel
-  {
-    using RAJA::internal::thread_privatize;
-    auto body = thread_privatize(loop_body);
-    forall_impl(InnerPolicy{}, std::forward<Iterable>(iter), body.get_priv());
-  }
+
+  RAJA::region<RAJA::omp_parallel_region>([=, &loop_body](){
```

Comment:
@trws , @ajkunen . I was able to fix the bugs, but in order to achieve correct results I have to capture loop_body by reference. If I don't, this branch fails certain unit tests in Kernel.cpp.